main.py

```diff
@@ -37,118 +96,49 @@ def apply_distance_calc(x, lat, lon):
-        # MAX_DISTANCE is divided by 2, since you'll need to come back to starting location
-        if distances_df.min().values[0] > (MAX_DISTANCE / 2):
-            print(
-                f"Sorry, no breweries within {MAX_DISTANCE}km from this starting location."
-            )
-            print(f"\nProgram took: {time.perf_counter() - start_time} seconds")
-            break
+def get_next_location(distances_df, km_left):
+    distances_df["distance_to_home"] = distances_df.apply(
+        apply_distance_calc, args=(start_lat, start_lon), axis=1
+    )
+    distances_df["ratio"] = distances_df.apply(calculate_ratio, axis=1)
+    best_next_location = pd.Series(distances_df.query("distance < 1000").ratio.idxmax())
+    if distances_df.distance_to_home.loc[best_next_location].values[0] > km_left:
+        return pd.Series(dtype="float64")
+    else:
+        return distances_df.loc[best_next_location]
 
-        # Count the ratio to identify best next brewery to visit
-        distances_df = pd.merge(
-            distances_df,
-            geocodes_df.beer_count,
-            left_index=True,
-            right_index=True,
-            how="inner",
-        )
-        distances_df["ratio"] = distances_df.apply(
-            lambda row: (row.beer_count + 1) / (row.distances + 0.01), axis=1
-        )
-        best_brewery_id = distances_df.query("distances < 1000").ratio.idxmax()
-
-        # Update distance variables
-        lat, lon = geocodes_df.loc[best_brewery_id, ["latitude", "longitude"]]
-        distance_to_start = calculate_distance(start_lat, start_lon, lat, lon)
-        current_distance += distances_df.distances.loc[best_brewery_id]
-        total_distance = current_distance + distance_to_start
-
-        # If total distance does not exceed the limit, add current location to the list
-        # Otherwise, finish the loop and adress variables for the results
-        if total_distance < MAX_DISTANCE or travel_df.empty:
-            brewery = pd.Series(
-                [best_brewery_id, distances_df.distances[best_brewery_id]],
-                index=travel_df.columns,
-            )
-            travel_df = travel_df.append(brewery, ignore_index=True)
-            geocodes_df.loc[best_brewery_id, "Visited"] = True
-        else:
-            lat, lon = geocodes_df.loc[
-                travel_df.brewery_id.tail(1).values[0], ["latitude", "longitude"]
-            ]
-            distance_to_start = calculate_distance(start_lat, start_lon, lat, lon)
-            current_distance -= distances_df.distances.loc[best_brewery_id]
-            total_distance = current_distance + distance_to_start
-            break
 
-    if not travel_df.empty:
-        # Print out the results
-        travel_df = travel_df.join(geocodes_df[["name", "latitude", "longitude", "beer_count"]], on="brewery_id", how="inner")
+def report(travel_df, start_time):
+    """
+    Print out the results
+    """
```

Comment:
For thought, report should be different from program logging what it is doing.
As we spoke, report should be something separated into its own data object, so that it could be moved easily.
These "print" lines would should be replaced with generation of string or some other data object that only would be printed once.

Reply:
Changed